.travis.yml

```diff
@@ -19,23 +20,56 @@ matrix:
   allow_failures:
     - php: hhvm
     - php: nightly
+    - php: 7.4
   fast_finish: true
 
 sudo: false
 
 install:
+  # Setup environment variables for testing
   - export AWS_ACCESS_KEY_ID=foo
   - export AWS_SECRET_ACCESS_KEY=bar
   - export AWS_CSM_ENABLED=false
-  - 'if [ $(phpenv version-name) == "5.5" ]; then rm ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini; fi'
-  - 'if [ $(phpenv version-name) != "hhvm" ] && [ $(phpenv version-name) != "nightly" ]; then echo "xdebug.overload_var_dump = 0" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi'
-  - 'if { [ $(phpenv version-name) == "7.2" ] || [ $(phpenv version-name) == "7.3" ]; } && [ -v COMPOSER_OPTS ]; then composer require --dev phpunit/phpunit "^5.7.11"; fi'
+
+  # Build job specific tweaks
+  - >
+    if [ $(phpenv version-name) == "5.5" ]; then
+      rm ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini
+    fi
+
+  - >
+    if [ $(phpenv version-name) != "hhvm" ] && [ $(phpenv version-name) != "nightly" ]; then
+      echo "xdebug.overload_var_dump = 0" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+    fi
+
+  - >
+    if { [ $(phpenv version-name) == "7.2" ] || [ $(phpenv version-name) == "7.3" ]; } && [ -v COMPOSER_OPTS ]; then
+      composer require --dev phpunit/phpunit "^5.7.11"
+    fi
+
+  # Resolve dependencies
   - composer --version
   - travis_retry composer update $COMPOSER_OPTS --no-interaction --prefer-source
 
 script:
+  # Unit tests
   - make test
+
+  # Static analysis
+  - >
```

Comment:
Should this get moved into the Makefile so it can be run offline?  And update CONTRIBUTING.md too?

Reply:
I hesitate to put this in the Makefile since it requires installing composer packages that are not part of require-dev, and which will only work with PHP 7.1+. 

Will update `CONTRIBUTING.md` for new guidelines.